Week3/homework/3-step3.js

```diff
@@ -3,23 +3,53 @@
 // use a 'for' loop
 function repeatStringNumTimesWithFor(str, num) {
   // add your code here
-  return str;
+  var strOut = "";
+  for (var i = 0; i < num; i++) {
+    strOut = strOut + str;
+  }
+  return strOut;
 }
 
 console.log('for', repeatStringNumTimesWithFor('abc', 3));
 
 // use a 'while' loop
 function repeatStringNumTimesWithWhile(str, num) {
   // add your code here
-  return str;
+  var str = 0;
+
+  while (str < 3) {
+    str++;
+  }
+
+
+  // expected output: 3
+
 }
 
 console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 
 // use a 'do...while' loop
 function repeatStringNumTimesWithDoWhile(str, num) {
   // add your code here
-  return str;
+  var str = "";
+  var i = 0;
+
+  do {
+    i = i + 1;
+    str = str + i;
+  } while (i < 3);
+
```

Comment:
First, you didn't return anything, you should return, as in the "for" example, which you did correctly, the new strOut variable created using the loop.
Secondly, "str = str + i;" this makes no sense, for several reasons: first, you don't want to add the index itself to the string. Secondly, you don't want to alter the parameter _str_ given. :) You just want to create a new string, as you did correctly in the "for" example, and add the given string _str_ several times. _i_ should be used just as a way to loop, never used in creating the string itself.
In conclusion, the code should be really similar to the previous one, just adding the condition to stop at the end:
Initialize _i_ and _strOut_.
Loop:
_do {
    i = i + 1;
    strOut = strOut + str;
  } while (i < num);_

And, of course, return the _strOut_.